sdk/samples/iot/paho_iot_hub_twin_cbor_sample.c

```diff
@@ -0,0 +1,533 @@
+  az_result rc
+      = az_iot_hub_client_twin_parse_received_topic(&hub_client, topic_span, out_twin_response);
+  if (az_result_failed(rc))
+  {
+    IOT_SAMPLE_LOG_ERROR("Message from unknown topic: az_result return code 0x%08x.", rc);
+    IOT_SAMPLE_LOG_AZ_SPAN("Topic:", topic_span);
+    exit(rc);
+  }
+  IOT_SAMPLE_LOG_SUCCESS("Client received a valid topic response.");
+  IOT_SAMPLE_LOG_AZ_SPAN("Topic:", topic_span);
+  IOT_SAMPLE_LOG_AZ_SPAN("Payload as ASCII:", message_span);
+  IOT_SAMPLE_LOG_HEX(
+      "Payload as Hex Value:", az_span_ptr(message_span), az_span_size(message_span));
+  IOT_SAMPLE_LOG("Status: %d", out_twin_response->status);
+}
+
+static void handle_device_twin_message(
+    MQTTClient_message const* message,
+    az_iot_hub_client_twin_response const* twin_response)
+{
+  // Invoke appropriate action per response type (3 types only).
+  switch (twin_response->response_type)
+  {
+    case AZ_IOT_HUB_CLIENT_TWIN_RESPONSE_TYPE_GET:
+      IOT_SAMPLE_LOG("Message Type: GET");
+      break;
+
+    case AZ_IOT_HUB_CLIENT_TWIN_RESPONSE_TYPE_REPORTED_PROPERTIES:
+      IOT_SAMPLE_LOG("Message Type: Reported Properties");
+      break;
+
+    case AZ_IOT_HUB_CLIENT_TWIN_RESPONSE_TYPE_DESIRED_PROPERTIES:
+      IOT_SAMPLE_LOG("Message Type: Desired Properties");
+
+      // Parse for the device count property.
+      int64_t desired_device_count;
+      if (parse_desired_device_count_property(message, &desired_device_count))
+      {
+        IOT_SAMPLE_LOG(" "); // Formatting
+
+        // Update device count locally and report update to server.
+        update_device_count_property(desired_device_count);
+        send_reported_property();
+        (void)receive_device_twin_message();
+      }
+      break;
+  }
+}
+
+static bool parse_desired_device_count_property(
+    MQTTClient_message const* message,
+    int64_t* out_parsed_device_count)
+{
+  bool property_found = false;
+  *out_parsed_device_count = 0;
+
+  // Parse message_span.
+  CborParser cbor_parser;
+  CborValue root;
+  CborValue desired_device_count;
+
+  // WARNING: Check the return of all API calls when developing your solution. Many Return checks
+  //          are ommited from this sample for simplification.
+
+  (void)cbor_parser_init(
+      (uint8_t*)message->payload, (size_t)message->payloadlen, 0, &cbor_parser, &root);
+  if (cbor_value_map_find_value(&root, desired_device_count_property_name, &desired_device_count)
+      == CborNoError)
+  {
+    if (cbor_value_is_integer(&desired_device_count))
+    {
+      (void)cbor_value_get_int64(&desired_device_count, out_parsed_device_count);
+      property_found = true;
+    }
+  }
+
+  if (property_found)
+  {
+    IOT_SAMPLE_LOG(
+        "Parsed desired `%s`: %" PRIi64,
+        desired_device_count_property_name,
+        *out_parsed_device_count);
+  }
+  else
+  {
+    IOT_SAMPLE_LOG(
+        "`%s` property was not found in desired property response.",
```

Comment:
nitnit: desired in this case is a message not a response. 
```suggestion
        "`%s` property was not found in desired property message.",
```

Reply:
will change in reg twin sample too.